ssl/quic/quic_rx_depack.c

```diff
@@ -0,0 +1,645 @@
+            /* FALLTHRU */
+        case OSSL_QUIC_FRAME_TYPE_PADDING:
+            /*
+             * We ignore this.
+             * For the PING frame, eliciting an ACK is enough.
+             * Since the PING frame is only a header, we simply
+             * skip it.
+             */
+            ossl_quic_wire_skip_frame_header(pkt, &frame_type);
+            break;
+
+        case OSSL_QUIC_FRAME_TYPE_ACK_WITHOUT_ECN:
+        case OSSL_QUIC_FRAME_TYPE_ACK_WITH_ECN: {
+            OSSL_QUIC_FRAME_ACK ack;
+            OSSL_QUIC_ACK_RANGE *ack_ranges;
+            uint64_t total_ranges = 0;
+            uint32_t ack_delay_exp = GET_CONN_ACK_DELAY_EXP(connection);
+            int ok = 1;          /* Assume the best */
+
+            if (!ossl_quic_wire_peek_frame_ack_num_ranges(pkt, &total_ranges)
+                || (ack_ranges = OPENSSL_zalloc(sizeof(ack_ranges[0])
+                                                * total_ranges)))
+                return 0;
+
+            ack.ack_ranges = ack_ranges;
+            ack.num_ack_ranges = total_ranges;
+
+            if (!ossl_quic_wire_decode_frame_ack(pkt, ack_delay_exp, &ack,
+                                                 NULL))
+                ok = 0;
+            if (ok
+                && mode == process_frames
+                && !ossl_ackm_on_rx_ack_frame(GET_CONN_ACKM(connection), &ack,
+                                              packet_space, received))
+                ok = 0;
+
+            OPENSSL_free(ack_ranges);
+            if (!ok)
+                return 0;
+            break;
+            }
+
+        case OSSL_QUIC_FRAME_TYPE_RESET_STREAM: {
+            OSSL_QUIC_FRAME_RESET_STREAM frame_data;
+
+            if (!ossl_quic_wire_decode_frame_reset_stream(pkt, &frame_data))
+                return 0;
+
+            switch (mode) {
+            case collect_ack_data:
+                /* This frame makes the packet ACK eliciting */
+                ackm_data->is_ack_eliciting = 1;
+                break;
+            case process_frames: {
+                QUIC_STREAM *stream
+                    = ssl_get_stream(connection, frame_data.stream_id);
+                int stream_type = ssl_get_stream_type(stream);
+
+                ssl_close_stream(stream);
+                if ((stream_type & SSL_STREAM_TYPE_W) != 0) {
+                    OSSL_STATM *statm = GET_CONN_STATM(connection);
+
+                    ssl_close_connection(connection);
+                    ossl_statm_set_error_state(statm, QUIC_STREAM_STATE_ERROR);
+
+                    /*
+                     * There's no point to continue with this packet.  However,
+                     * processing it was a success, so we return with a happy
+                     * face, and leave it to the state machine what to do with
+                     * the general STREAM_STATE_ERROR state.
+                     */
+                    return 1;
+                }
+                break;
+                }
```

Comment:
Although it might look like a bikeshedding issue to outsiders I think we will need a coding-style policy amendment proposal and an OTC vote to resolve this.

Reply:
And yeah, this is still ugly although a little less than the original formatting.

Reply:
> Although it might look like a bikeshedding issue to outsiders ...

... not just outsiders :wink:
(I'm too passionate not to participate, mind you)

Reply:
> And yeah, this is still ugly although a little less than the original formatting.

Thing is that the switch body is hard to get right / consistent, it's a fairly strange construct as the language goes, and many battles have been made over the best way to lay it out.  This will probably not be the last...

Had C been able to have variable declarations interspersed with statements from the very start, we wouldn't have had this discussion...

Me, I'm starting to think that I should put all these "offending" code blocks into separate functions and simply call them.  That would take away the "offending" braces and end this discussion as well.

Reply:
That would work for me, and have the benefit of making the function a lot smaller.

Reply:
> Me, I'm starting to think that I should put all these "offending" code blocks into separate functions and simply call them. That would take away the "offending" braces and end this discussion as well.

I've decided to go that way.
End of bikeshedding.